relay-general/src/store/transactions/processor.rs

```diff
@@ -350,19 +350,29 @@ impl Processor for TransactionsProcessor<'_> {
         if self.name_config.scrub_identifiers {
             // Normalize transaction names for URLs and Sanitized transaction sources.
             // This in addition to renaming rules can catch some high cardinality parts.
+            let mut sanitize_transaction = false;
+
             if matches!(
                 event.get_transaction_source(),
                 &TransactionSource::Url | &TransactionSource::Sanitized
             ) {
-                scrub_identifiers(&mut event.transaction)?;
+                let scrubbed_identifiers = scrub_identifiers(&mut event.transaction);
+                if scrubbed_identifiers.map_or(false, |scrubbed| scrubbed) {
```

Comment:
AFAIK we still want to early return the error here if `scrub_identifiers` returns one.

```suggestion
                let scrubbed_identifiers = scrub_identifiers(&mut event.transaction)?;
                if scrubbed_identifiers {
```